src/Lecture3.hs

```diff
@@ -208,10 +233,28 @@ types that can have such an instance.
 -}
 
 -- instance Foldable Weekday where
+    -- There is no logical operation for folding over an @Enum@, it'd be like somehow folding over an @Int@.
 -- instance Foldable Gold where
+    -- Same as before, @Gold@ is just a newtype around @Int@
 -- instance Foldable Reward where
--- instance Foldable List1 where
--- instance Foldable Treasure where
+    -- Same again. This time it's a wrapper around two values, but the same principle applies
+
+-- A @List1@ is essentially like @NonEmpty@, so it gets its @Foldable@ instance abusing the fact that it's essentially a list.
+instance Foldable List1 where
+    foldr :: (a -> b -> b) -> b -> List1 a -> b
+    foldr f z (x `List1` xs) = f x $ foldr f z xs
+
+    foldMap :: Monoid m => (a -> m) -> List1 a -> m
+    foldMap f = foldr ((<>) . f) mempty
```

Comment:
That's already the default implementation of `foldMap` so there's not much benefit in implementing it explicitly in the exactly the same way 🙂 
But maybe you can think of a different implementation 🤔